reconcile/utils/jjb_client.py

```diff
@@ -399,3 +396,21 @@ def get_trigger_phrases_regex(job: dict) -> str | None:
             if "github-pull-request" in trigger:
                 return trigger["github-pull-request"].get("trigger-phrase")
         return None
+
+    @staticmethod
+    def get_gitlab_webhook_trigger(job: dict) -> list[str] | None:
+        gitlab_triggers = job["triggers"][0]["gitlab"]
+        # pr-check job should be triggered by merge request events
+        # and certain comments: [test]|/retest|/lgtm|/lgtm cancel|/hold|/hold cancel
+        if gitlab_triggers.get("trigger-merge-request"):
+            trigger = ["mr", "note"]
+        # build main/master job should be triggered by push events
+        elif gitlab_triggers.get("trigger-push"):
+            trigger = ["push"]
+        # On-demand test job should be triggered by special comment
+        elif gitlab_triggers.get("trigger-note"):
+            trigger = ["note"]
+        else:
+            trigger = None
+
+        return trigger
```

Comment:
it's better to keep return value as not `None` to match https://github.com/app-sre/qontract-reconcile/pull/4770/files#diff-f910de274ff6f61d8f3549e1e2a965487e59bf2f7ee8466b6a77659962357dd1R27-R33, then need to update `if trigger is None:` -> `if not trigger:`. This can also make `in` type check easier as `trigger` is always a list.

And can simplify condition by early return

```python
        if gitlab_triggers.get("trigger-merge-request"):
            return ["mr", "note"]
        if gitlab_triggers.get("trigger-push"):
            return ["push"]
        if gitlab_triggers.get("trigger-note"):
            return ["note"]
        return []
```